modules/analysis/src/endpoints/tests/mod.rs

```diff
@@ -335,6 +333,50 @@ async fn quarkus_component_by_purl(ctx: &TrustifyContext) -> Result<(), anyhow::
     Ok(())
 }
 
+/// find a component by query
+#[test_context(TrustifyContext)]
+#[test(actix_web::test)]
+async fn find_component_by_query(ctx: &TrustifyContext) -> Result<(), anyhow::Error> {
+    ctx.ingest_documents(["spdx/quarkus-bom-3.2.11.Final-redhat-00001.json"])
+        .await?;
+
+    // NOTE: Testing for qualified purls is tricky, because the order
+    // of the qualifiers isn't predictable. One workaround is to "and"
+    // the qualifiers in the query using the LIKE operator,
+    // e.g. q=purl~BASE&purl~QUALIFIER_ONE&purl~QUALIFIER_TWO
+    const PURL: &str = "pkg:maven/com.redhat.quarkus.platform/quarkus-bom@3.2.11.Final-redhat-00001?repository_url=https://maven.repository.redhat.com/ga/&type=pom";
+
+    async fn by_query(app: &impl CallService, query: &str) -> Value {
+        let uri = format!("/api/v2/analysis/component?q={}&limit=0", encode(query));
+        let request: Request = TestRequest::get().uri(&uri).to_request();
+        let response: Value = app.call_and_read_body_json(request).await;
+        tracing::debug!(test = "", "{response:#?}");
+        response
+    }
+
+    let app = caller(ctx).await?;
+
+    for each in [
+        "purl=pkg:maven/com.redhat.quarkus.platform/quarkus-bom@3.2.11.Final-redhat-00001?type=pom\\&repository_url=https%3a%2f%2fmaven.repository.redhat.com%2fga%2f",
+        "purl~pkg:maven/com.redhat.quarkus.platform/quarkus-bom@3.2.11.Final-redhat-00001&purl~type=pom&purl~repository_url=https%3A%2F%2Fmaven.repository.redhat.com%2Fga%2F",
+        "purl~quarkus-bom",
+        "cpe=cpe:/a:redhat:quarkus:3.2::el8",
+        "cpe~cpe:/a:redhat:quarkus:3.2::el8",
+        "cpe~cpe:/a:redhat:quarkus:3.2",
+        "cpe~cpe:/a::quarkus",
+        "cpe~redhat",
```

Comment:
@JimFuller-RedHat I have these queries working now. Please let me know if I missed any that you need.